smf/switch_zone_setup/switch_zone_setup

```diff
@@ -1,19 +1,32 @@
-#!/bin/bash
+#!/bin/ksh
+
 set -ex -o pipefail
 
 . /lib/svc/share/smf_include.sh
 
-# set up the users required for wicket.
-USERS=(wicket support)
+# set up the users required for wicket and support.
+USERS=(
+    (user=wicket group=wicket gecos='Wicket User')
+    (user=support group=support gecos='Oxide Support'
+        profiles=('Primary Administrator')
+    )
+)
+
+for i in "${!USERS[@]}"; do
+    nameref u=USERS[$i]
 
-for user in ${USERS[@]}; do
-    if ! id -u $user >/dev/null 2>&1; then
-        # Add a new group for the user.
-        groupadd "$user"
-        # Add the user.
-        useradd -m -g "$user" "$user"
-        # Remove the password from the user to enable passwordless login.
-        passwd -d "$user"
+    # Add a new group for the user.
+    getent group "${u.group}" >/dev/null 2>&1 || groupadd "${u.group}"
+    # Add the user.
+    getent passwd "${u.user}" >/dev/null 2>&1 \
+        || useradd -m -g "${u.group}" -c "${u.gecos}" "${u.user}"
+    # Remove the password from the user to enable passwordless login.
+    passwd -d "${u.user}"
+    # Assign or remove profiles
+    if [[ -n "${u.profiles}" ]]; then
+        usermod -P"$(printf '%s,' "${u.profiles}")" "${u.user}"
```

Comment:
Why does this use `printf` instead of `-P"${u.profiles}"`?

Reply:
The ksh printf builtin implicitly joins arrays, so this is a way to convert an array of strings into a comma-delimited string. I missed an `[@]` there which didn't matter since there is just one profile, but I've fixed it now. Here's an example:

```
build% cat test
#!/bin/ksh

USERS=(
    (user=support group=support gecos='Oxide Support'
        profiles=('Service Operator' 'Hotplug Management' 'Zone Management')
    )
)

for i in "${!USERS[@]}"; do
    nameref u=USERS[$i]

    echo `printf '%s,' "${u.profiles[@]}"`
done

build% ./test
Service Operator,Hotplug Management,Zone Management,
```